libs/island-ui/core/src/lib/PhoneInput/PhoneInput.tsx

```diff
@@ -0,0 +1,340 @@
+                <label
+                  htmlFor={id}
+                  className={cn(styles.label, styles.labelSizes[size], {
+                    [styles.labelDisabledEmptyInput]:
+                      disabled && !value && !defaultValue,
+                  })}
+                >
+                  {label}
+                  {required && (
+                    <span aria-hidden="true" className={styles.isRequiredStar}>
+                      {' '}
+                      *
+                    </span>
+                  )}
+                  {tooltip && (
+                    <Box marginLeft={1} display="inlineBlock">
+                      <Tooltip text={tooltip} />
+                    </Box>
+                  )}
+                </label>
+              )}
+              <Box display="flex">
+                <CountryCodeSelect
+                  id={selectId}
+                  name={selectId}
+                  onChange={handleSelectChange}
+                  value={selectedCountryCode}
+                  options={countryCodes}
+                  disabled={disabled || readOnly}
+                  backgroundColor={backgroundColor}
+                  inputHasLabel={!!label}
+                  size={size}
+                  onFocus={() => setHasFocus(true)}
+                  onBlur={() => setHasFocus(false)}
+                  onMenuOpen={() => setIsMenuOpen(true)}
+                  onMenuClose={() => setIsMenuOpen(false)}
+                  dataTestId={`country-code-test-${id}`}
+                />
+                <NumberFormat
+                  className={cn(
+                    styles.input,
+                    resolveResponsiveProp(
+                      backgroundColor,
+                      styles.inputBackgroundXs,
+                      styles.inputBackgroundSm,
+                      styles.inputBackgroundMd,
+                      styles.inputBackgroundLg,
+                      styles.inputBackgroundXl,
+                    ),
+                    styles.inputSize[size],
+                  )}
+                  id={id}
+                  name={name}
+                  type="tel"
+                  disabled={disabled}
+                  getInputRef={mergedRefs}
+                  placeholder={placeholder}
+                  value={value?.replace(cc, '')}
+                  defaultValue={getDefaultValue(
+                    defaultValue,
+                    defaultCountryCode,
+                  )}
+                  readOnly={readOnly}
+                  format={
+                    countryCodeList.find(
+                      (x) => x.dial_code === cc && !!x.format,
+                    )?.format
+                  }
+                  onValueChange={({ value }) => {
+                    // Don't prefix value with country code it it's empty.
+                    value
+                      ? onFormatValueChange?.(cc + value)
+                      : onFormatValueChange?.(value)
+                  }}
```

Comment:
It's not clear to me how PhoneInput combines the country code and the phone number into the onChange event handler. Does NumberFormat handle it with this hook?

Does it trim all extra symbols? Let's say I have US +1, (123) 32-12321 or something; Is the value +11233212321?